### PR TITLE
Add Admin::Accounts methods

### LINKED ISSUE #323: Implement Admin Accounts API (Part 1/2)
Mastodon's Admin Accounts API allows an admin to perform moderation action with accounts. The goal of this task is to implement the following endpoints of this API in BigBone:

- View accounts (v1)
- View accounts (v2)
- View a specific account
- Approve a pending account
- Reject a pending account

The Admin Accounts API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/admin/accounts/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### LINKED ISSUE #324: Implement Admin Accounts API (Part 2/2)
Mastodon's Admin Accounts API allows an admin to perform moderation action with accounts. The goal of this task is to implement the following endpoints of this API in BigBone:

- Delete an account
- Perform an action against an account
- Enable a currently disabled account
- Unsilence an account
- Unsuspend an account
- Unmark an account as sensitive

The Admin Accounts API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/admin/accounts/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

This PR implements the methods available via the [Admin::Accounts](https://docs.joinmastodon.org/methods/admin/accounts/) endpoints. Despite the branch saying `part-1`, this actually implements all of them. After having started, I no longer even checked which ones were part of the ticket.

> [!WARNING]
> I have not implemented the v1 variant as it’s due for removal next year anyway.

> [!NOTE]
> [Mastodon API documentation examples](https://docs.joinmastodon.org/methods/admin/accounts/) still contain `ip` as an object instead of a String. I have decided to implement `AdminAccount` by how it’s defined in the [Admin::Account documentation](https://docs.joinmastodon.org/entities/Admin_Account/#ip) instead.

Closes #323, closes #324.

# Type of Change

- New feature
- Documentation

# Breaking Changes

- None.

# How Has This Been Tested?

New unit tests for all methods added including their optional parameter variants

# Mandatory Checklist

- [x] I ran `gradle check` and there were no errors reported
- [ ] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [x] I have added KDoc documentation to all public methods

# Optional checks

- [x] In case you worked on a new feature: Did you also implement the reactive endpoint (bigbone-rx)?
- [x] In case you added new `*Methods` classes: Did you also reference it in the `MastodonClient` main class?
- [x] Did you also update the documentation in the `/docs` folder (e.g. API Coverage page)?

### CODE REVIEW

File: bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminAccountMethods.kt [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+     * @param displayName Search for the given display name
+     * @param byDomain Filter by the given domain
+     * @param emailAddress Lookup a user with this email
+     * @param ipAddress Lookup users with this IP address
+     *
+     * @see <a href="https://docs.joinmastodon.org/methods/admin/accounts/#v2">Mastodon API documentation: admin/accounts/#v2</a>
+     */
+    @JvmOverloads
+    fun viewAccounts(
+        range: Range = Range(),
```
Comment (PattaFeuFeu) [inline bigbone-rx/src/main/kotlin/social/bigbone/rx/admin/RxAdminAccountMethods.kt:44]: I’ve moved `range` way up here as all other parameters are optional and thus Java users would have to use the variant with all parameters even if they only wanted to use the `range` parameter.

File: bigbone/src/main/kotlin/social/bigbone/api/entity/Role.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    @SerialName("updated_at")
+    @Serializable(with = DateTimeSerializer::class)
+    val updatedAt: PrecisionDateTime = PrecisionDateTime.InvalidPrecisionDateTime.Unavailable
+) {
+
+    /**
+     * Returns whether this role grants the [Permission] specified in [permission].
+     * The Mastodon API returns permissions as a bitmask in [rawPermissions].
+     */
+    @Suppress("DataClassContainsFunctions")
```
Comment (PattaFeuFeu) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/Role.kt:69]: I’ve added this suppression as I don’t agree with detekt’s defaults here.
I want these essentially conversion methods to be limited to this class’ instances only so I think the current way is the best we could do.

File: bigbone/src/main/kotlin/social/bigbone/api/entity/Role.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    fun getParsedPermissions(): List<Permission> = Permission.entries.filter(::hasPermission)
+
+    /**
+     * Possible Permission Flag values masked in [rawPermissions].
+     *
+     * Use [hasPermission] to check for a specific permission’s availability, or [getParsedPermissions] to get a list
+     * of [Permission]s.
+     */
+    enum class Permission(val bitValue: Int) {
+        Administrator(0b1), // 1
```
Comment (PattaFeuFeu) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/Role.kt:87]: I’ve used the binary variants of writing `Int` here to help people reading the code to understand what is happening in `hasPermission` which utilises a bit-wise `and`.

The commented values are those we would see if we printed them from what we get via the API.

File: bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminAccount.kt [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+     * Whether the account is currently approved.
+     */
+    @SerialName("approved")
+    val approved: Boolean,
+
+    /**
+     * Whether these account’s post are currently flagged sensitive.
+     */
+    @SerialName("sensitized")
+    val sensitized: Boolean? = null,
```
Comment (PattaFeuFeu) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/admin/AdminAccount.kt:100]: Not mentioned in documentation but available in data returned by some methods, so I added it here, but as a nullable value with null default.

File: bigbone/src/main/kotlin/social/bigbone/api/entity/Role.kt [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        ManageRules(0b1000000000000), // 0x1000 / 4096
+        ManageAnnouncements(0b10000000000000), // 0x2000 / 8192
+        ManageCustomEmojis(0b100000000000000), // 0x4000 / 16384
+        ManageWebhooks(0b1000000000000000), // 0x8000 / 32768
+        ManageRoles(0b100000000000000000), // 0x20000 / 131072
+        ManageUserAccess(0b1000000000000000000), // 0x40000 / 262144
+
+        InviteUsers(0b10000000000000000), // 0x10000 / 65536
+
+        DeleteUserData(0b10000000000000000000) // 0x80000 / 524288
```
Comment (PattaFeuFeu) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/Role.kt:132]: I’ve used the binary variants of writing Int here to help people reading the code to understand what is happening in `hasPermission` and `getParsedPermission` which utilise a bit-wise `and`.

The commented values are those we get [in the documentation](https://docs.joinmastodon.org/entities/Role/#permission-flags) (hex) and those we would see if we printed them from what we get via the API.

**Review by bocops — APPROVED** [review]
Code looks good, I'm fine with everything you highlighted via comment.

I'm a bit short on time, so I didn't check the new assets in detail, but assume that those were just copied from available examples and are OK.

Would it be possible to refactor the apiName definition in the enums somehow, so that we don't have to repeat them that often? Doesn't need to happen in this PR, of course, but if at all possible, a new issue could be opened for that.

**Comment by PattaFeuFeu** [comment]
@bocops Thanks! If `Role` is alright with you in the current state, I can use the definition of this PR and use it in #411 as well 😊 

I’ll create an issue regarding `apiName` to check. Might be possible to have an extension function that only works on enum classes tagged with `Serializable`.

**Comment by PattaFeuFeu** [comment]
@bocops Created: https://github.com/andregasser/bigbone/issues/412

**Pull request merged**